flink-formats/flink-json/pom.xml

```diff
@@ -70,17 +69,24 @@ under the License.
 
 		<dependency>
 			<groupId>org.apache.flink</groupId>
-			<!-- use a dedicated Scala version to not depend on it -->
-			<artifactId>flink-table_2.11</artifactId>
+			<artifactId>flink-table_${scala.binary.version}</artifactId>
 			<version>${project.version}</version>
 			<type>test-jar</type>
 			<scope>test</scope>
 		</dependency>
 
-		<!-- flink-table needs Scala -->
+		<!-- flink-table needs these -->
 		<dependency>
 			<groupId>org.scala-lang</groupId>
-			<artifactId>scala-compiler</artifactId>
+			<artifactId>scala-library</artifactId>
+			<scope>test</scope>
+		</dependency>
+		<!-- the reason things normally work out without this is that this is a transitive
```

Comment:
this is a weird comment; doesn't that apply to all undeclared usages of transitive dependencies?